htlcswitch/hop/iterator.go

```diff
@@ -116,14 +170,32 @@ func (r *sphinxHopIterator) HopPayload() (*Payload, error) {
 			bytes.NewReader(r.processedPacket.Payload.Payload),
 		)
 		if err != nil {
-			return nil, err
+			// If we couldn't even parse our payload then we do
+			// a best-effort of determining our role in a blinded
+			// route, accepting that we can't know whether we
+			// were the introduction node (as the payload
+			// is not parseable).
+			routeRole := RouteRoleCleartext
+			if r.blindingKit.UpdateAddBlinding.IsSome() {
+				routeRole = RouteRoleRelaying
+			}
+
+			return nil, routeRole, err
 		}
 
+		// Now that we've parsed our payload we can determine which
+		// role we're playing in the route.
+		_, payloadBlinding := parsed[record.BlindingPointOnionType]
+		routeRole := NewRouteRole(
```

Comment:
Updated the comment on this function to explain, summary of converstion:
* We can only _get_ our payload blinding point if we've received a valid `update_add_htlc` blinding point (otherwise, we can't unwrap the onion properly)
* If we've received a valid `update_add_htlc` blinding point, then we're a relaying node because the _preceeding_ node must have received a valid blinding point to be able to give us the next one
* If we then *also* get a payload blinding point, it must be the sender who was buggy
* We'll fail back with a malformed error, which the node before us will handle - this is fine in mission control, and even if we're penalized